test/consul_helpers.go

```diff
@@ -0,0 +1,197 @@
+package test
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/gruntwork-io/terratest/modules/gcp"
+	"github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/gruntwork-io/terratest/modules/retry"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/gruntwork-io/terratest/modules/test-structure"
+	"github.com/hashicorp/consul/api"
+)
+
+// RepoRoot represents the root of the project.
+const RepoRoot = "../"
+
+const ConsulClusterExampleVarProject = "gcp_project"
+const ConsulClusterExampleVarZone = "gcp_zone"
+const ConsulClusterExampleVarServerClusterName = "consul_server_cluster_name"
+const ConsulClusterExampleVarClientClusterName = "consul_client_cluster_name"
+
+const ConsulClusterExampleVarServerClusterTagName = "consul_server_cluster_tag_name"
+const ConsulClusterExampleVarClientClusterTagName = "consul_client_cluster_tag_name"
+
+const ConsulClusterExampleVarServerSourceImage = "consul_server_source_image"
+const ConsulClusterExampleVarClientSourceImage = "consul_client_source_image"
+
+const ConsulClusterExampleVarServerClusterSize = "consul_server_cluster_size"
+const ConsulClusterExampleVarClientClusterSize = "consul_client_cluster_size"
+
+const ConsulClusterExampleDefaultNumServers = 3
+const ConsulClusterExampleDefaultNumClients = 4
+
+const ConsulClusterExampleOutputServerInstanceGroupName = "instance_group_name"
+const ConsulClusterExampleOutputClientInstanceGroupName = "client_instance_group_name"
+
+// SavedGCPZone represents the key to use when saving the GCP zone.
+const SavedGCPZone = "GCPZone"
+
+const GoogleProjectIDEnvVar = "GOOGLE_CLOUD_PROJECT_ID"
+
+// Test the consul-cluster example by:
+//
+// 1. Copying the code in this repo to a temp folder so tests on the Terraform code can run in parallel without the
+//    state files overwriting each other.
+// 2. Building the Image in the consul-image example with the given build name
+// 3. Deploying that Image using the consul-cluster Terraform code
+// 4. Checking that the Consul cluster comes up within a reasonable time period and can respond to requests
+func runConsulClusterTest(t *testing.T, packerBuildName string, examplesFolder string, packerTemplatePath string) {
+	exampleFolder := test_structure.CopyTerraformFolderToTemp(t, RepoRoot, examplesFolder)
+
+	test_structure.RunTestStage(t, "setup_image", func() {
+
+		// Get the Project Id to use
+		gcpProjectID := gcp.GetGoogleProjectIDFromEnvVar()
```

Comment:
We might use a better way to handle Google credentials, but I'll leave this here for now.